File: Sources/BeagleUI/Sources/DeclarativeViews/Models/UIComponents/NavigationBar.swift
//
//  NavigationBar.swift
//  BeagleUI
//
//  Created by Gabriela Coelho on 18/11/19.
//  Copyright © 2019 Zup IT. All rights reserved.
//

public struct NavigationBar: NativeWidget {
    
    // MARK: - Public Properties
    
    public let title: String
    public let leading: Widget?
    public let trailing: Widget?
    
    // MARK: - Initialization

    init(
        title: String,
        leading: Widget? = nil,
        trailing: Widget? = nil
    ) {
        self.title = title
        self.leading = leading
        self.trailing = trailing
    }

    public init(
         title: String,
         @WidgetBuilder leading leadingBuilder: () -> Widget,
         @WidgetBuilder trailing trailingBuilder: () -> Widget
    ) {
        self.title = title
        self.leading = leadingBuilder()
        self.trailing = trailingBuilder()
    }
}

File: src/main/kotlin/br/com/zup/beagleui/framework/widget/ui/NavigationBar.kt
package br.com.zup.beagleui.framework.widget.ui

import br.com.zup.beagleui.framework.widget.core.NativeWidget
import br.com.zup.beagleui.framework.widget.core.Widget

data class NavigationBar(
    val title: String,
    val leading: Widget? = null,
    val trailing: Widget? = null
) : NativeWidget